torch/_dynamo/symbolic_convert.py

```diff
@@ -314,6 +314,10 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
             if truth_fn(eval_result):
                 push and self.push(value)
                 self.jump(inst)
+        elif isinstance(value, TensorVariable) and value.dtype == torch.bool:
+            if truth_fn(value.get_real_value()):
+                push and self.push(value)
+                self.jump(inst)
```

Comment:
* If the value is tensor, it has been handled at L248, I don't think this code snippet will be executed.
* You bake the value during compilation time, if the value is updated from ```tensor(True)``` to ```tensor(False)``` in the next iteration, guard won't be failed and you will get wrong result.

I'm curious the motivation behind this change, since we have already handle if cond on tensor at L248.

Reply:
> If the value is tensor, it has been handled at L248, I don't think this code snippet will be executed.

This is for export though, so self.should_compile_partial_graph() = False` because `self.one_graph = True`

Reply:
> You bake the value during compilation time, if the value is updated from tensor(True) to tensor(False) in the next iteration, guard won't be failed and you will get wrong result.

Ohh, I see. Is there a way for me to add a guard that `torch.all(x) == True` ?